src/pages/ProfilesExplorerView/components/SceneExploreServiceLabels/components/SceneGroupByLabels/SceneGroupByLabels.tsx

```diff
@@ -274,20 +273,18 @@ export class SceneGroupByLabels extends SceneObjectBase<SceneGroupByLabelsState>
   openExpandedPanelDrawer(item: GridItemData) {
     const serviceName = getSceneVariableValue(this, 'serviceName');
     const profileMetricId = getSceneVariableValue(this, 'profileMetricId');
+    const title = `${serviceName} · ${getProfileMetricLabel(profileMetricId)} · ${item.label}`;
+
+    const headerActions = () => [new SelectAction({ EventClass: EventSelectLabel, item }), new FavAction({ item })];
 
     this.state.drawer.open({
-      title: `${serviceName} · ${getProfileMetricLabel(profileMetricId)} · ${item.label}`,
-      body:
-        item.panelType === PanelType.BARGAUGE
-          ? new SceneLabelValuesBarGauge({
-              item,
-              headerActions: () => [new SelectAction({ EventClass: EventSelectLabel, item }), new FavAction({ item })],
-            })
-          : new SceneLabelValuesTimeseries({
-              displayAllValues: true,
-              item,
-              headerActions: () => [new SelectAction({ EventClass: EventSelectLabel, item }), new FavAction({ item })],
-            }),
+      title,
+      body: panelBuilder(item.panelType, {
+        displayAllValues: true,
+        legendPlacement: 'right',
```

Comment:
Moving the placement of the legend when opened in the drawer (we have more width).